prow/prstatus/prstatus.go

```diff
@@ -176,14 +177,28 @@ func (da *DashboardAgent) HandleUserDashboard(queryHandler PullRequestQueryHandl
 				serverError("Error with getting user login", err)
 				return
 			}
-			// Saves login
+			// Saves login. We saves the login under 2 cookies. One for the use of client to render the
+			// data and one encoded for server to verify the identity of the authenticated user.
+			http.SetCookie(w, &http.Cookie{Name: "github_login", Value: login})
 			session.Values[loginKey] = login
 			if err := session.Save(r, w); err != nil {
 				serverError("Save oauth session", err)
 				return
 			}
 			// Construct query
 			query := da.ConstructSearchQuery(login)
+			if err := r.ParseForm(); err == nil {
+				if q := r.Form.Get("query"); q != "" {
+					query = q
+				}
+			}
+			// If repo is not specified in the search query. We limit the search to repos that are
+			// configured with either Prow or Tide.
+			if !specifiedRepo(query) {
+				for _, v := range da.repos {
+					query += fmt.Sprintf(" repo:\"%s\"", v)
+				}
+			}
```

Comment:
I think the API should not mutate the query string in any way, we should return an informative error, and the client should not submit invalid queries. We can easily handle this client side.

Reply:
I think we have a problem that the list of repos configured with prow is very long. Users may not be able to remember all repos in the case they want to know all possible prs. Also, we cannot list all of them in the URL as well, so I think it is better to do the work for them if users don't specify repos.

Reply:
How can I write my own queries then if I can't see what query is actually being run ... ?

Reply:
For example, I want to query the PRs in test-infra that are open and not LGTM yet, I should be able to do this, but I can't if you silently insert a list of repos in the backend...

Reply:
Oh I see, this is only the case that repos aren't specified.

I still think this is magical and that we'll need to clearly document this somewhere. It should be clear what can and cannot be queried.

Reply:
Yes, I think I will but a question mark /help icon declaring that if you dont specify your repo. We are going to limit them by ones that configured with Prow.

Reply:
If we do that we need to do this same thing for `is:pr`

Reply:
@cjwagner @BenTheElder  To enforce the `is:pr` query, I think we can append `is:pr` to the end of each search query. This will override any `is:pr` or `is:issue` before and make sure the query is looking for pull requests